### ISSUE TITLE
nxos_logging - Feature to add source-interface and use-vrf 

### ISSUE THREAD

**Issue by e2-team** [ansible/ansible#43089]
##### SUMMARY

Need to have a playbook that deploy our syslog server configuration in all our Nexus.
We have been able to do this with nxos_command plugin, but we want to do some checking before putting that config in place. That's why we prefer to have the nxos_logging.
I saw that now we have some "server" function added lately, that's a big plus. But from my understanding we are missing some configuration for that to work properly.
We would need the following configuration:
```
logging server <SYSLOG_SERVER_IP> 5 use-vrf management
logging source-interface mgmt0
```
And now we are only getting:
```
logging server <SYSLOG_SERVER_IP>
```

##### ISSUE TYPE

 - Add functionality for source-interface
 - Add functionality for user-vrf 

##### COMPONENT NAME

nxos_logging

##### ANSIBLE VERSION

```
ansible 2.6.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Apr 11 2018, 07:36:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### CONFIGURATION

No configurations changed in ansible.cfg

##### OS / ENVIRONMENT
Ansible Server: CentOS Linux release 7.5
Device: Nexus N9K with version 7.0(3)I2(2a)

##### STEPS TO REPRODUCE

```yaml
  tasks:
  - name: Configure syslog in Nexus
    nxos_logging:
       dest: server
       remote_server: "{{ server }}"
       name: "{{ server }}"
       facility_level: 5

```

##### EXPECTED RESULTS

```
logging server <SYSLOG_SERVER_IP> 5 use-vrf management
logging source-interface mgmt0
```
##### ACTUAL RESULTS

```
logging server <SYSLOG_SERVER_IP>
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/network/nxos/nxos_logging.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_logging.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mikewiebe @privateip @rahushen @rcarrillocruz @trishnaguha @tstoner
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by e2-team** [comment]
Hi everyone, I was able to do what I need by hard-coding part of the commands missing into the python module in my ansible server. 
But would be great to have something to set source-interface and using use-vrf. I saw that modules like ntp do the source-interface. So I think we can check the code over there. 
I'm not an expert in python, so to get this done would take me too much time, if some one else can help me would be great.
